tooling/nargo_toml/src/lib.rs

```diff
@@ -47,6 +47,35 @@ pub fn find_file_manifest(current_path: &Path) -> Option<PathBuf> {
 }
 
 /// Returns the [PathBuf] of the directory containing the `Nargo.toml` by searching from `current_path` to the root of its [Path].
+/// When `workspace` is `true` it returns the topmost directory, when `false` the innermost one.
+///
+/// Returns a [ManifestError] if no parent directories of `current_path` contain a manifest file.
+pub fn find_root(current_path: &Path, workspace: bool) -> Result<PathBuf, ManifestError> {
+    if workspace {
+        find_package_root(current_path)
+    } else {
+        find_file_root(current_path)
+    }
+}
```

Comment:
This is great and an improvement over the current behaviour but there's still some holes where someone can have a workspace which sits within another workspace.

If someone tries to run a command within the inner workspace they'll end up with very confusing errors messages rather than it complaining about nested workspaces. We can handle this in a followup PR as it's low priority.